### PR TITLE
[ADC-513] Vue Integration

### PR DESCRIPTION
**Pull request by vasswann**
adCAPTCHA Vue integration

### CODE REVIEW

File: examples/node/.env [hunk 1]
```diff
@@ -1 +1 @@
-API_KEY=
+API_KEY= "835e7672-4754-423b-b583-e99ab8d28021"
```
Comment (Lucasarenaterry) [inline examples/node/.env]: Change this so its in a .env.local

and put "Your api key here"

File: examples/vue/.gitignore [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+.idea
+.DS_Store
+*.suo
+*.ntvs*
+*.njsproj
+*.sln
+*.sw?
+# Ignore .env files containing sensitive data
+.env.local
+.env.*.local
```
Comment (Lucasarenaterry) [inline examples/vue/.gitignore]: add empty line at the bottom

File: examples/vue/src/assets/background.jpg
Comment (Lucasarenaterry) [inline examples/vue/src/assets/background.jpg:1]: could you change this to use the adcaptcha logo instead:

https://www.notion.so/adcaptcha/67d0195ee5944561a932923e9a61eefd?v=cede034db8794a6c9439000a0a1765a0&pvs=4

File: examples/vue/src/components/ContactForm.vue [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    padding: 0.6rem; 
+    font-size: 0.8rem; 
+  }
+
+  .form-button {
+    padding: 0.6rem; 
+    font-size: 0.9rem; 
+  }
+}
+</style>
```
Comment (Lucasarenaterry) [inline examples/vue/src/components/ContactForm.vue]: Add empty line

File: examples/vue/src/components/Footer.vue [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    max-width: 1200px;
+    margin: 0 auto; 
+    padding: 0 1rem;
+  }
+  
+  .copyright {
+    font-size: 1rem;
+    margin-bottom: 1rem;
+  }
+  </style>
```
Comment (Lucasarenaterry) [inline examples/vue/src/components/Footer.vue]: Add empty line

File: examples/vue/src/components/HeroSection.vue [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  .hero-button {
+    padding: 0.5rem 1rem;
+    font-size: 1rem;
+  }
+
+  .hero-content {
+    padding: 1.5rem;
+  }
+}
+  </style>
```
Comment (Lucasarenaterry) [inline examples/vue/src/components/HeroSection.vue]: Add empty line

File: examples/vue/src/components/Navbar.vue [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+.navbar-links a:hover {
+  color: #e63946; 
+}
+
+.navbar-scrolled {
+  background-color: rgba(8, 7, 7, 0.9); 
+}
+  </style>
```
Comment (Lucasarenaterry) [inline examples/vue/src/components/Navbar.vue]: Add empty line

File: examples/vue/src/components/ResponseMessage.vue [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+</script>
+  
+<style scoped>
+  .response-message {
+    color: #e63946; 
+    font-size: 1.1rem;
+    margin-bottom: 1rem;
+    text-align: center;
+  }
+  </style>
```
Comment (Lucasarenaterry) [inline examples/vue/src/components/ResponseMessage.vue]: Add empty line

File: examples/vue/src/style.css [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+  padding: 0;
+  min-width: 320px;
+  min-height: 100vh;
+  overflow-x: hidden;
+  box-sizing: border-box;
+}
+
+html {
+  scroll-behavior: smooth;
+}
```
Comment (Lucasarenaterry) [inline examples/vue/src/style.css]: Add empty line

File: packages/vue/src/components/adCAPTCHA.vue [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+        console.error('Error loading adCAPTCHA script:', err);
+      });
+    });
+
+    return {
+      placementID: props.placementID,
+    };
+  },
+});
+</script>
```
Comment (Lucasarenaterry) [inline packages/vue/src/components/adCAPTCHA.vue]: Add empty line here

File: packages/vue/src/index.tsx [hunk 10]
```diff
@@ -0,0 +1,6 @@
+// src/index.tsx
+
+import AdCAPTCHA from './components/adCAPTCHA.vue';  
+import { setKeywords, getSuccessToken } from './util'; 
+export { AdCAPTCHA, setKeywords, getSuccessToken };
+export default AdCAPTCHA;
```
Comment (Lucasarenaterry) [inline packages/vue/src/index.tsx]: Remove comment and add empty line to the bottom

**Review by Lucasarenaterry — CHANGES_REQUESTED** [review]

File: examples/node/.env (hunk 1 above)
Comment (vasswann) [reply examples/node/.env]: I created .env.local and the api key been removed from .env

File: examples/vue/src/assets/background.jpg
Comment (vasswann) [reply examples/vue/src/assets/background.jpg:1]: The image is used for background.  As we talked about it, it's remained the same.

File: examples/vue/.gitignore (hunk 2 above)
Comment (Lucasarenaterry) [inline examples/vue/.gitignore]: .env.local
.env.*.local

These are not needed as you already have *.local

**Review by Lucasarenaterry — APPROVED** [review]

File: examples/vue/.gitignore (hunk 2 above)
Comment (vasswann) [reply examples/vue/.gitignore]: .gitignore update is done

File: examples/vue/src/assets/background.jpg
Comment (MattGibney) [inline examples/vue/src/assets/background.jpg:1]: Remove this. The examples needs to be professional and on-brand with adCAPTCHA.

File: examples/vue/src/components/Footer.vue
Comment (MattGibney) [inline examples/vue/src/components/Footer.vue:1]: We don't really need this, especially not a copyright.

File: examples/vue/src/assets/background.jpg
Comment (MattGibney) [inline examples/vue/src/assets/background.jpg:1]: Remove this. Example styling needs to be brand consistent.

**Review by MattGibney — CHANGES_REQUESTED** [review]

File: examples/vue/src/assets/background.jpg
Comment (vasswann) [reply examples/vue/src/assets/background.jpg:1]: Background image removed, adCAPTCHA logo has added to the example page
Screenshot 2024-11-19 at 10 28 37

File: examples/vue/src/components/Footer.vue
Comment (vasswann) [reply examples/vue/src/components/Footer.vue:1]: Footer component removed

File: examples/vue/src/assets/background.jpg
Comment (vasswann) [reply examples/vue/src/assets/background.jpg:1]: Image has been removed.

**Review by MattGibney — APPROVED** [review]

**Pull request merged**